utils/globals.py

```diff
@@ -30,7 +33,63 @@ def set_lab(lab):
 def get_lab():
     if _LAB is None:
         modlog.error('get_lab called before set_lab')
-        #modlog.error('An unexpected error occurd')re
         sys.exit(1)
     return _LAB
 
+
+def set_sampler(sampler_name, version):
+    ''' sets the name and version of the sampler as the sampler UID
+
+    :param sampler_name: name of the method used for samplng
+    :param version: version of the method used for sampling reported
+                    by the sampling function
+    :returns: **no returns, call func get_sampler_uid
+    '''
+    global _SAMPLER, _SAMPLER_has_been_set, _SAMPLER_UID
+
+    _SAMPLER = sampler_name
+    _SAMPLERVERSION = version
+    _SAMPLER_UID = 'escalate_' + str(sampler_name) + '_' + str(version)
+
+    _SAMPLER_has_been_set = True
+
+
+def get_sampler_uid():
+    '''returns the sampler used for automated sampling
+    '''
+    global _SAMPLER_UID
+    if _SAMPLER is None:
+        modlog.error('get_sampler_uid called before set_sampler')
+        sys.exit(1)
+    return _SAMPLER_UID
+
+
+def get_sampler_author():
+    import devconfig
+    sampler_author = 'escalate_' + str(devconfig.RoboVersion)
+    return(sampler_author)
+
+
+def get_user_author_name():
+    message = ("Enter the name you would like to appear for manual submissions (e.g., Ian Pendleton):")
+    user_input = input(message)
+
+    with open('./capture/user_cli_variables.py', 'w') as fout:
+        fout.write("# This file acts as a local cache to the user's preferred author name\n")
+        fout.write("user_author_name = '%s'\n" % user_input)
+    # import the repo_path from the python file cache we just created
+    from capture.user_cli_variables import user_author_name
+    return user_author_name
+
```

Comment:
This seems quite messy to do something in the style that we're already doing it in this very file (saving something such that it can be accessed by any function in the code.)

We're creating a local file to hold the author's name. However, we will have to do so every time this fn is called. 

Why not use the `standard globals` logic to store it as a variable instead of as a file.

Reply:
Because there is no location to currently input that information.  Also, this way we don't have to have the user enter their name each time that they run the code.  Once the code is run it stores their name for the rest of the time they have that version of the code.